### PR TITLE
Feature/1204 application search bar

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Impl. the basic app search logic.

### CODE REVIEW

**Comment by kolloch** [comment]
Hey @philipnrmn, can you shepherd this PR? Thanks a lot.

**Comment by aldipower** [comment]
@kolloch I will care..

File: src/main/resources/assets/js/components/AppListComponent.jsx [hunk 1]
```diff
@@ -15,6 +15,10 @@ var AppListComponent = React.createClass({
     router: React.PropTypes.object.isRequired
   },
 
+  getInitialState: function() {
+    return  {filter:''};
```
Comment (aldipower) [inline src/main/resources/assets/js/components/AppListComponent.jsx]:
```
return {filter: ""};
```

File: src/main/resources/assets/js/components/AppListComponent.jsx [hunk 2]
```diff
@@ -29,23 +33,36 @@ var AppListComponent = React.createClass({
     collection.sort();
   },
 
+  filterCollection: function (filter) {
+    this.setState({filter:filter||''});
```
Comment (aldipower) [inline src/main/resources/assets/js/components/AppListComponent.jsx]: I wouldn't handle the fallback to "" at this point, would check it where it is needed.

File: src/main/resources/assets/js/components/AppListComponent.jsx [hunk 3]
```diff
@@ -29,23 +33,36 @@ var AppListComponent = React.createClass({
...
+  },
+
   getAppNodes: function () {
+    // Set the app filter pattern
+    var pattern = new RegExp("^/?"+this.state.filter);
```
Comment (aldipower) [inline src/main/resources/assets/js/components/AppListComponent.jsx]: The string has to be escaped, so ```\``` and other chars can not be used.

File: src/main/resources/assets/js/components/AppListComponent.jsx [hunk 4]
```diff
@@ -29,23 +33,36 @@ var AppListComponent = React.createClass({
...
-        /* jshint trailing:false, quotmark:false, newcap:false */
-        /* jscs:disable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
-        return (
-          <AppComponent
-            key={model.id}
-            model={model}
-            router={this.props.router} />
-        );
+        //  Render app nodes based on app filter
+        if(pattern.test(model.id)){
```
Comment (aldipower) [inline src/main/resources/assets/js/components/AppListComponent.jsx]: Cool!

File: src/main/resources/assets/js/components/AppListComponent.jsx [hunk 5]
```diff
@@ -29,23 +33,36 @@ var AppListComponent = React.createClass({
...
+              router={this.props.router} />
+          );
+        }
         /* jshint trailing:true, quotmark:true, newcap:true */
         /* jscs:enable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
       }, this)
     );
   },
 
+  componentWillReceiveProps: function(props) {
```
Comment (aldipower) [inline src/main/resources/assets/js/components/AppListComponent.jsx]: You should add the ```props.filter``` to the defaultProps.

File: src/main/resources/assets/js/components/FilterComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+var React = require("react/addons");
+
+var FilterComponent = React.createClass({
+  displayName: "FilterComponent",
+  propTypes: {
+    onChange: React.PropTypes.func
+  },
+
+  getInitialState: function() {
+    return  {filter:null};
```
Comment (aldipower) [inline src/main/resources/assets/js/components/FilterComponent.jsx]:
```
return {filter: ""};
```

File: src/main/resources/assets/js/components/FilterComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    return { onChange: function () {} };
+  },
+
+  onInputChange: function (event) {
+    var value = this.refs.filterInput.getDOMNode().value;
+    this.updateFilter(value);
+  },
+
+  updateFilter:function(value){
+
```
Comment (aldipower) [inline src/main/resources/assets/js/components/FilterComponent.jsx]: Remove empty line

File: src/main/resources/assets/js/components/FilterComponent.jsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    var id = this.props.id;
+    var label = this.props.label;
+    var placeholder = this.props.placeholder;
+
+    return (
+      <div class="form-group"  >
+        <label >{label}
+        <input type="text"
+                 ref="filterInput"
```
Comment (aldipower) [inline src/main/resources/assets/js/components/FilterComponent.jsx]: Idendation

File: src/main/resources/assets/js/components/FilterComponent.jsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                className="form-control"
+                onChange={this.onInputChange}
+                placeholder={placeholder}
+                value={this.state.filter} />
+                </label>
+        <button type="button"
+                className="btn btn-default"
```
Comment (aldipower) [inline src/main/resources/assets/js/components/FilterComponent.jsx]: Indendation

```
<button type="button"
    className="btn btn-default"
    onClick={this.clearFilter}>
  Clear
</button>
```

File: src/main/resources/assets/js/components/Marathon.jsx [hunk 10]
```diff
@@ -408,6 +411,10 @@ var Marathon = React.createClass({
     });
   },
 
+  onAppFilterChange:function (value) {
+    this.setState({appFilter:value})
```
Comment (aldipower) [inline src/main/resources/assets/js/components/Marathon.jsx]: Missing semicolon

File: src/main/resources/assets/js/components/Marathon.jsx [hunk 11]
```diff
@@ -408,6 +411,10 @@ var Marathon = React.createClass({
     });
   },
 
+  onAppFilterChange:function (value) {
```
Comment (aldipower) [inline src/main/resources/assets/js/components/Marathon.jsx]: Missing space

File: src/main/resources/assets/js/components/Marathon.jsx [hunk 12]
```diff
@@ -471,9 +480,13 @@ var Marathon = React.createClass({
           <a href="#newapp" className="btn btn-success navbar-btn" >
             + New App
           </a>
+          <FilterComponent
+              placeholder="Filter list"
```
Comment (aldipower) [inline src/main/resources/assets/js/components/Marathon.jsx]: Too much idendation

File: src/main/resources/assets/js/components/Marathon.jsx [hunk 13]
```diff
@@ -490,6 +503,10 @@ var Marathon = React.createClass({
     /* jscs:enable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
   },
 
+
```
Comment (aldipower) [inline src/main/resources/assets/js/components/Marathon.jsx]: Remove unnecessary lines

**Comment by aldipower** [comment]
Looks pretty good for 3 hours! :)

**Comment by kolloch** [comment]
@aldipower Is this the same as #1542? Can we close this?

**Comment by aldipower** [comment]
Workday PR. Closing this now, can we reopened if necessary.

**Pull request closed without merging**